src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.NoSuchElementException;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int fruitsInStorage = Storage.storage.get(fruit);
```

Comment:
Instead of directly accessing the storage map, you should use a DAO layer to encapsulate the storage operations. This will adhere to the encapsulation principle and make your code more maintainable.